doc/man3/OSSL_HTTP_parse_url.pod

```diff
@@ -0,0 +1,74 @@
+=pod
+
+=head1 NAME
+
+OSSL_HTTP_parse_url,
+OCSP_parse_url
+- http utility functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/http.h>
+
+ int OSSL_HTTP_parse_url(const char *url,
+                         int *pssl, char **puser, char **phost,
+                         char **pport, int *pport_num,
+                         char **ppath, char **pquery, char **pfrag);
+
+Deprecated since OpenSSL 3.0, can be hidden entirely by defining
+B<OPENSSL_API_COMPAT> with a suitable version value, see
+L<openssl_user_macros(7)>:
+
+ int OCSP_parse_url(const char *url, char **phost, char **pport, char **ppath,
+                    int *pssl);
+
+=head1 DESCRIPTION
+
+OSSL_HTTP_parse_url() parses its input string I<url> as a URL of the form
+C<[http[s]://][userinfo@]host[:port][/path[?query]][#fragment]> and splits it up
```

Comment:
This is slightly incorrect from a standards standpoint, but what's more, this is not what I see your code doing.  To be perfectly correct (and what the function seems to do), that should be:

`C<[http[s]://][userinfo@]host[:port][/path][?query][#fragment]>`

In other words, the query part isn't *part of* the path.

Reply:
> `C<[http[s]://][userinfo@]host[:port][/path][?query][#fragment]>`

It should be `/[path]` since there must be a slash after the hostpart. Or just `/path` and indicate that it can be an empty string.

I also think it's a mistake to make the protocol be optional. Even though the function has `http` in the name, you're allowing both http and https, and it's forward-compatible for when you add other URI types.

I also think, and maybe not in this manpage or PR, you need to make *very clear* that https is not supported. It should be an error if the client tries to use https.

Reply:
Good catch!
I had stated the pattern this way in order to reflect that the client side can see the query component as part of the path.
This morning I thought that this pattern is not really correct, but then decided to leave it in for now, 
as a kind of Easter Egg to be discovered by a careful reviewer - and inevitably you found it :+1: 
Fixed.

Reply:
glad you're fixing the pattern.  and the other two comments?